test/e2e/e2e_test.go

```diff
@@ -87,6 +89,64 @@ var _ = Describe("E2E", func() {
 				// TODO Check that the expected CRDs, deployments, services, etc. are present
 			})
 		})
+		Context("Mixerless Telemetry Stats Filter Test", func() {
+			// var resourcesFile string
+			var (
+				istio        v1beta1.Istio
+				majorMinor   string
+			)
+			const timeout = 30 * time.Second
+			const interval = 1 * time.Second
+
+			JustBeforeEach(func() {
+				// get the istio object created in the OUTER JustBeforeEach
+				log.Info("Namespace: ", "Namespace", instance.Namespace)
+				log.Info("Name: ", "Name", instance.Name)
+				Expect(GetIstioObject(&istio, instance.Namespace, instance.Name)).Should(Succeed())
+				version := istio.Spec.Version // get first 2 digits
+				versionParts := strings.SplitN(string(version), ".", 3)
+				majorMinor = fmt.Sprintf("%s.%s", versionParts[0], versionParts[1])
+				log.Info("Istio Version: ", "Version", majorMinor)
+			})
+
+			It("Transitions Filter through all states", func() {
```

Comment:
I don't think that transitions actually matters in these tests, so it doesn't matter what was the previous state, it doesn't change the actual state.

Because of that, these could be written as three separate tests in my mind. When the enabled field is `true`, `false` and `nil`. Also, in my mind these should be separately runnable tests existing on their own, so they shouldn't be run after each other but be different small/reproducible test cases.

Reply:
1. The cluster is torn down and re-created between tests, so this would add 2-3 minutes per test, so we opted for only 1 test.
2. All states are tested, and the transitions between states are also tested. From a tester/customer perspective, a running cluster needs to be able to be changed without complete tear down and re-creation, so the transitions become important to the customer.

Reply:
Of course, I agree with the first part, the cluster should NOT be torn down and recreated between test cases.

In my mind, there should be an initial setup, which brings the cluster to a state, which is a starting point for all test cases. Then each test case should do something and at the end clean up after itself so that the original starting point state is recreated and the next test case can go from there. (So not the whole cluster is torn down, only the initial state is recreated.)

So honestly, I'd do this with these mixerlesstelemetry test cases as well. I'm not convinced by your second point that it's worth testing the transitions this way. You also say that "a running cluster needs to be able to be changed without complete tear down and re-creation", which I agree with, but still these could be separate test cases in my mind as I suggested earlier without the complete tear down.

I might be wrong of course and I also don't want to hold this PR up even more, so if you insist on this solution it is fine by me.

Reply:
Changed the comments in the test.
spreading out the states to individual tests increased run time by 2 minutes.